src/main/java/step2/StringAddCalculator.java

```diff
@@ -0,0 +1,48 @@
+package step2;
+
+import step2.delimiter.Delimiter;
+import step2.delimiter.DelimiterFactory;
+import step2.exception.StringAddIllegalArgumentException;
+
+public class StringAddCalculator {
+    private static final DelimiterFactory delimiterFactory = new DelimiterFactory();
+
+    public static int splitAndSum(String text) {
+        if (isEmpty(text)) {
+            return 0;
+        }
+        Delimiter delimiter = delimiterFactory.getDelimiter(text);
+        String[] splitedString = delimiter.splitFrom(text);
+        return calculateSum(splitedString);
+    }
+
+    private static boolean isEmpty(String text) {
+        return text == null || text.isBlank();
+    }
+
+    private static int calculateSum(String[] stringArr) {
+        int sum = 0;
+        for (String each : stringArr) {
+            int num = parseIntFrom(each);
+            validateIsNotNegative(num);
+            sum += num;
+        }
+        return sum;
+    }
+
+    private static int parseIntFrom(String s) {
+        try {
+            return Integer.parseInt(s);
+        } catch (NumberFormatException e) {
+            return 0;
+        }
+    }
+
+    private static void validateIsNotNegative(int num) {
```

Comment:
```suggestion
    private static void validatePositive(int num) {
```
부정표현대신 양수를 판별한다는 의미로 네이밍하는 게 더 자연스러워보이는데 어떻게 생각하시나요?

Reply:
음수인지 판단하는 조건을 고민없이 그대로 메소드명으로 옮겨버렸네요. 피드백 주신 내용과 같이 Positive로 작성하는 게 훨씬 명확하게 느껴집니다. 앞으로 미션 간 부정어구 표현을 최대한 피해보겠습니다 감사합니다!